spec/shared-examples/subprocess-runner.rb

```diff
@@ -0,0 +1,71 @@
+shared_examples_for "a subprocess runner" do |fixture_root|
+
+  describe "running '/bin/echo test'" do
+    subject { described_class.new(%w[/bin/echo test]) }
+
+    it "sets the exit code to 0" do
+      result = subject.run
+      expect(result.exit_code).to eq 0
+    end
+
+    it "returns the contents of stdout" do
+      result = subject.run
+      expect(result.stdout).to eq 'test'
+    end
+
+    it "indicates the command succeeded" do
+      result = subject.run
+      expect(result).to be_success
+      expect(result).to_not be_failed
+    end
+  end
+
+  describe "running '/bin/ls' with a different working directory" do
+    subject do
+      described_class.new(%w[/bin/ls]).tap do |o|
+        o.cwd = fixture_root
+      end
+    end
+
+    it "returns the contents of the given working directory" do
+      result = subject.run
+      expect(result.stdout).to eq 'no-execute.sh'
+    end
+  end
+
+  describe "running '/bin/false'" do
+    subject { described_class.new(%w[/bin/false]) }
+
+    it "sets the exit code to 1" do
+      result = subject.run
+      expect(result.exit_code).to eq 1
```

Comment:
I suspect that this will make the tests fail on your system, which will be a problem. I think that we could use use the PATH variable instead of a fully qualified binary.